src/pointer-analysis/goto_program_dereference.h

```diff
@@ -65,9 +65,12 @@ class goto_program_dereferencet:protected dereference_callbackt
 
   const symbolt *get_or_create_failed_symbol(const exprt &expr) override;
 
+  DEPRECATED(SINCE(2019, 05, 22, "use list returning version instead"))
   void
   get_value_set(const exprt &expr, value_setst::valuest &dest) const override;
 
+  std::list<exprt> get_value_set(const exprt &expr) const override;
```

Comment:
If we're adding a new entry-point we probably shouldn't use `std::list`, there's no call for a doubly-linked list here AFAICT

Reply:
but it is a list, and the fact that `value_setst::valuest` is a list is public, by writing `value_setst::valuest` we are just making it more complicated for the reader

Reply:
I'm not suggesting `valuest`, I'm suggesting `std::vector` because that's the most performant C++ sequence container when you don't have a good reason for something else (e.g. deque or list)

Reply:
oh right